### PR TITLE
New IIO C-API to read configuration attributes

### PR DESCRIPTION
**Pull request by guchaojie**
Add 3 C-API to read scale, offset and sampling_frequency attribute.

Signed-off-by: guchaojie chao.jie.gu@intel.com

### CODE REVIEW

**Comment by guchaojie** [comment]
@ceolin @edersondisouza @fulong82 @laykuanloon  This PR is for issue #2117. All thie C-API is to get configuration attribute from device level.  Please have  a review, any comment is welcome.

File: src/lib/io/include/sol-iio.h [hunk 1]
```diff
@@ -249,6 +249,58 @@ int
...
+char *sol_iio_channel_get_name(struct sol_iio_channel *channel);
+
+
+/**
+ * @brief Gets the configuration scale attribute from one device
+ *
+ * This function gets configuration scale attribute from one device,
+ * after the device sol_iio_device structure has been desired.
+ *
+ * @param device The sol_iio_device structure which the device is desired
```
Comment (ceolin) [inline src/lib/io/include/sol-iio.h]: which the scale is desired ...

File: src/lib/io/include/sol-iio.h [hunk 2]
```diff
@@ -249,6 +249,58 @@ int
...
+ *
+ */
+char *sol_iio_channel_get_name(struct sol_iio_channel *channel);
+
+
+/**
+ * @brief Gets the configuration scale attribute from one device
+ *
+ * This function gets configuration scale attribute from one device,
+ * after the device sol_iio_device structure has been desired.
```
Comment (ceolin) [inline src/lib/io/include/sol-iio.h]: this paragraph is adding nothing, i think you can remove it.

File: src/lib/io/include/sol-iio.h [hunk 3]
```diff
@@ -249,6 +249,58 @@ int
...
+ *
+ * @return 0 on success, -errno on failure.
+ */
+int sol_iio_device_get_scale(struct sol_iio_device *device, const char *prefix_name, double *scale);
+
+/**
+ * @brief Gets the configuration offset attribute from one device
+ *
+ * This function gets configuration offset attribute from one device,
+ * after the device sol_iio_device structure has been desired.
```
Comment (ceolin) [inline src/lib/io/include/sol-iio.h]: ditto

File: src/lib/io/sol-iio.c [hunk 4]
```diff
@@ -571,14 +571,16 @@ setup_device_reader(struct sol_iio_device *device)
...
-channel_get_pure_name(struct sol_iio_channel *channel)
+channel_get_pure_name(char *name)
 {
     char channel_name[NAME_MAX];
     size_t channel_name_len;
     char *channel_pure_name;
     bool modified = false;
 
-    strncpy(channel_name, channel->name, sizeof(channel_name) - 1);
+    SOL_NULL_CHECK(name, NULL);
```
Comment (ceolin) [inline src/lib/io/sol-iio.c]: I've spotted this problem in another commit :)

File: src/lib/io/sol-iio.c [hunk 5]
```diff
@@ -1762,3 +1764,108 @@ sol_iio_mount_calibration(struct sol_iio_device *device, sol_direction_vector *v
...
+            device->device_id, _name)) { \
+            r = sol_util_read_file(path, "%lf", scale); \
+            goto end; \
+        } \
+    } while (0)
+
+    GET_SCALE(prefix_name);
+    pure_name = channel_get_pure_name(prefix_name);
+    GET_SCALE(pure_name);
+
```
Comment (ceolin) [inline src/lib/io/sol-iio.c]: this code is almost the same of `channel_get_scale()`. Wouldn't be better both use same code ? You can put this code in another function and both call it. 

One more thing, the scale is not associated with a channel ? if yes, is it possible get it directly from the desired channel looking the channel in the channels vector ?

File: src/modules/flow/iio/iio.json [hunk 6]
```diff
@@ -308,6 +368,21 @@
...
+        },
+        {
+         "data_type": "float",
+         "description": "Offset to be added to device raw readings",
+        "name": "OFFSET"
+        },
+        {
+         "data_type": "int",
+         "description": "Sampling frequency of the sensor. If -1, use device default",
+         "name": "SAMPLING_FREQ"
```
Comment (ceolin) [inline src/modules/flow/iio/iio.json]: since we are using frequency everywhere, I guess it would be better `SAMPLIGING_FREQUENCY` don't you think ?

**Comment by ceolin** [comment]
patch looks good, just check whether my comments make sense or not :)

File: src/lib/io/sol-iio.c (hunk 5 above)
Comment (guchaojie) [inline src/lib/io/sol-iio.c]: the scale in base function is not assiociated with a channel.

**Pull request closed without merging**